source/server/hot_restart.proto

```diff
@@ -38,10 +38,19 @@ message HotRestartMessage {
       uint64 memory_allocated = 1;
       uint64 num_connections = 2;
 
-      // The parent's current values for various stats in its stats store.
       // Keys are fully qualified stat names.
-      map<string, uint64> counters = 3;
-      map<string, uint64> gauges = 4;
+      //
+      // The parent's total accumulated value of the counter. The hot restart parent should only
+      // ever send a given counter in this field once. It is expected that there will also be an
+      // entry in counter_deltas for the same counter name. The value in counter_values should be
+      // counter.value() - (the value stored in counter_deltas, i.e. counter.latch()).
+      map<string, uint64> counter_values = 3;
```

Comment:
One thing for discussion: I think we can skip the complexity around sending values at all. IMO it's OK for the counters to reset to 0 in the new process. All stats backends should deal with this correctly. WDYT?

/wait-any

Reply:
Nice, didn't realize that was ok. That should certainly be cleaner.

Reply:
Done.